core/bin/external_node/src/main.rs

```diff
@@ -387,46 +379,77 @@ async fn main() -> anyhow::Result<()> {
-                    L1ExecutedBatchesRevert::Allowed,
-                );
-                reverter
-                    .rollback_db(last_correct_batch, BlockReverterFlags::all())
-                    .await;
-                tracing::info!("Rollback successfully completed, the node has to restart to continue working");
-                return Ok(());
-            } else {
-                tracing::error!("Reorg detector actor failed");
+
+    let wait_for_tasks = wait_for_tasks(
+        task_handles,
+        particular_crypto_alerts,
+        graceful_shutdown,
+        tasks_allowed_to_finish,
+    );
+
+    tokio::pin!(wait_for_tasks);
+
+    loop {
+        tokio::select! {
+            _ = &mut wait_for_tasks => {
+                tracing::info!("Tasks have stopped, shutting down");
+            },
+            _ = &mut sigint_receiver => {
+                tracing::info!("Stop signal received, shutting down");
+                stop_sender.send(true).ok();
+            },
+            consistency_checker = &mut consistency_checker_handle => {
+                // Consistency checker exited, but the reorg detector may still
+                // be doing its thing. Signal it that things should be stop but
+                // carry on the loop execution until the reorg checker actually stops.
+                match consistency_checker {
+                    Ok(Ok(())) => tracing::info!("Consistency Checker has stopped"),
+                    Ok(Err(error)) => tracing::error!("Consistency checker has stopped with {error}"),
+                    Err(error) => tracing::error!("Consistency checker handle has failed with {error}"),
+                }
+                stop_sender.send(true).ok();
+            },
+            last_correct_batch = &mut reorg_detector_handle => {
+                if let Ok(last_correct_batch) = last_correct_batch {
+                    if let Some(last_correct_batch) = last_correct_batch {
+                        tracing::info!("Performing rollback to block {}", last_correct_batch);
+                        shutdown_components(&stop_sender, health_check_handle).await;
+                        let reverter = BlockReverter::new(
+                            config.required.state_cache_path,
+                            config.required.merkle_tree_path,
+                            None,
+                            connection_pool,
+                            L1ExecutedBatchesRevert::Allowed,
+                        );
+                        reverter
+                            .rollback_db(last_correct_batch, BlockReverterFlags::all())
+                            .await;
+                        tracing::info!("Rollback successfully completed, the node has to restart to continue working");
+                    }
+                } else {
+                    tracing::error!("Reorg detector actor failed");
+                }
+                break
             }
+            else => break
```

Comment:
Stupid question: What does `else` do here? [Tokio docs say](https://docs.rs/tokio/latest/tokio/macro.select.html) that it's used with conditional branches, but there are no conditional branches in this `select!`.

Reply:
Yep should be removed.